api/envoy/extensions/filters/http/aws_request_signing/v3/aws_request_signing.proto

```diff
@@ -63,6 +75,20 @@ message AwsRequestSigning {
   // - exact: bar
   // When applied, all headers that start with "x-envoy" and headers "foo" and "bar" will not be signed.
   repeated type.matcher.v3.StringMatcher match_excluded_headers = 5;
+
+  // Optional Signing algorithm specifier, either aws_sigv4 or aws_sigv4a, defaulting to aws_sigv4.
+  SigningAlgorithm signing_algorithm = 6;
+
+  // A region set is a comma separated list of AWS regions, such as ``us-east-1,us-east-2`` or wildcard ``*``
+  // or even region strings containing wildcards such as ``us-east-*``
+  //
+  // By configuring a region set, a sigv4a signed request can be sent to multiple regions, rather than being
+  // valid for only a single region destination.
+  //
+  // region_set can only be specified if signing_algorithm = aws_sigv4a
+  //
+  // If region_set is unspecified and signing_algorithm = aws_sigv4a, the region_set will be taken from the aws profile
+  string region_set = 7;
```

Comment:
I basically prefer you previous design. A explicit option `signing_algorithm` is used to determine the content type of `region` field. I think may be we needn't this `region_set` here.

Reply:
signing_algorithm does not currently exist. so for backward compatibility I was intending to make it optional.

so my original design was signing_algorithm defaults to aws_sigv4. if signing_algorithm is specified and equal to aws_sigv4a then region is used as a region set. otherwise region is used as a region with the sigv4 algorithm

if this design works for you then I will implement it that way.